docs/guides/ingesters-with-wal.md

```diff
@@ -0,0 +1,75 @@
+---
+title: "Ingesters with WAL"
+linkTitle: "Ingesters with WAL"
+weight: 5
+slug: ingesters-with-wal
+---
+
+Currently the ingesters without the TSDB stores all the data in the memory and if it happens to crash for any reason, there will be a loss of data. WAL helps fill this gap of reliability.
+
+To use WAL, there are some changes that needs to be made in the deployment.
+
+## Things to change
+
+1. Since ingesters need to have the same persistent volume across restarts/rollout, all the ingesters should be run on [statefulset](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/) with fixed volumes.
+
+2. Following flags needs to be set
+    * `--ingester.wal-dir` to the directory where the WAL data should be stores and/or recovered from.
+    * `--ingester.wal-enabled` to `true` which enables writing to WAL during ingestion.
+    * `--ingester.checkpoint-enabled` to `true` to enable checkpointing of in-memory chunks to disk. This is optional which helps in speeding up the replay process.
+    * `--ingester.checkpoint-duration` to the interval at which checkpoints should be created.
```

Comment:
Can we have a recommendation here (what is the default?)

Reply:
The recommendation depends on the need and the ingester size. The default is 30m, will add that here, but we have 15m in our dev.